site/examples/check-lists.js

```diff
@@ -32,7 +32,39 @@ const CheckListsExample = () => {
 }
 
 const withChecklists = editor => {
-  const { deleteBackward } = editor
+  const { deleteBackward, insertBreak } = editor
+  editor.insertBreak = () => {
+    const match = Editor.above(editor, {
+      match: n => {
+        return Editor.isBlock(editor, n)
+      },
+    })
+    if (match) {
+      const [matchingNode] = match
+      if (matchingNode.type !== 'paragraph') {
```

Comment:
I'm not sure you need to handle the paragraph type here. The existing insertBreak handler you're calling at the end should be able to deal with this case handily.